artifacts/source_type.py

```diff
@@ -271,3 +271,20 @@ def __init__(self, query=None, base_object=None):
     super(WMIQuerySourceType, self).__init__()
     self.query = query
     self.base_object = base_object
+
+
+SOURCE_TYPES = {
```

Comment:
> The idea is that users of the library are able to register a new handler by accessing the module global. 

Are you using that approach somewhere for reference?

Also the idea of this project is to provide means to verify definitions, having private subclasses kind of defeats that purpose.